core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -725,6 +726,20 @@ class AutoTuner(
     }
   }
 
+  /**
+   * Recommendation for 'spark.rapids.file.cache' based on read characteristics of job.
+   */
+  private def recommendFileCache() {
+    if (appInfoProvider.isParquetOrOrc
+          && appInfoProvider.getDistinctLocationPct < DEF_DISTINCT_READ_THRESHOLD
+          && appInfoProvider.getTotalReadSize > DEF_READ_SIZE_THRESHOLD) {
```

Comment:
Not necessarily required for this PR, but it would be good if this size check was localized to the data sources that are being redundantly read rather than the aggregate read size of all tables.  Could be redundantly reading a very tiny table but only accessing a huge table once, and this seems like it will recommend caching.

Reply:
I changed to getRedundantReadSize, but the code looks ugly so let me know if it can be simplified.